arctic/store/_ndarray_store.py

```diff
@@ -145,9 +143,6 @@ def _index_range(self, version, symbol, from_version=None, **kwargs):
         """
         from_index = None
         if from_version:
-            if version['base_sha'] != from_version['base_sha']:
```

Comment:
Isn't this still a worry?  A `write` fighting an `append` might cause corruption?